core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -491,9 +491,13 @@ public synchronized void initialize(URI uri, org.apache.hadoop.conf.Configuratio
     // Create FileSystem for accessing Alluxio.
     // Disable URI validation for non-Alluxio schemes.
     boolean disableUriValidation =
-        (uri.getScheme() != null) ? uri.getScheme().equals(Constants.SCHEME) : true;
-    mFileSystem = FileSystem.Factory.create(
-        ClientContext.create(subject, alluxioConf).setUriValidationEnabled(disableUriValidation));
+        (uri.getScheme() == null) || uri.getScheme().equals(Constants.SCHEME);
+    if (alluxioConf.getBoolean(PropertyKey.USER_LOCAL_CACHE_LIBRARY)) {
+      mFileSystem = FileSystem.Factory.create(null, alluxioConf);
```

Comment:
Would it be okay now that the file system client does not have the client user subject?

Reply:
that shouldn't be necessary in library mode